File: InfoIsland/InfoIsland/src/lib/models/eachFeed.ts
export interface Feeds {
  feeds: EachFeed[];
  hadExceeded: boolean;
}

export interface EachFeed {
  item_id: string;
  site_url: string;
  title: string;
  description: string;
  feed_url: string;
  language?: Language;
  dt_created: string;
  dt_updated: string;
  favorites: number;
}

export enum Language {
  En = "en",
  EnUS = "en-US",
  EnUs = "en-us",
  Ja = "ja",
}

File: InsightStream/InsightStream/domain/baseFeeds/eachFeed.go
package baseFeeds

import (
	"github.com/google/uuid"
	"time"
)

type EachFeed struct {
	Id          uuid.UUID `json:"item_id"`
	SiteURL     string    `json:"site_url,omitempty"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	FeedURL     string    `json:"feed_url"`
	Language    string    `json:"language,omitempty"`
	DtCreated   time.Time `json:"dt_created"`
	DtUpdated   time.Time `json:"dt_updated"`
	Favorites   int       `json:"favorites"`
}
